qbft/spectest/generate/main.go

```diff
@@ -37,6 +41,59 @@ func main() {
 
 	fmt.Printf("found %d tests\n", len(all))
 	writeJson(byts)
+
+	for _, testF := range spectest.AllTests {
+		test := testF()
+		// generate post state comparison
+		post, err := test.GetPostState()
+		if err != nil {
+			panic(err.Error())
+		}
+		writeJsonStateComparison(test.TestName(), reflect.TypeOf(test).String(), post)
+	}
+}
+
+func clearStateComparisonFolder() {
+	_, basedir, _, ok := runtime.Caller(0)
+	if !ok {
+		panic("no caller info")
+	}
+	dir := filepath.Join(strings.TrimSuffix(basedir, "main.go"), "state_comparison")
+
+	if err := os.RemoveAll(dir); err != nil {
+		panic(err.Error())
+	}
+
+	if err := os.Mkdir(dir, 0700); err != nil {
+		panic(err.Error())
+	}
+}
+
+func writeJsonStateComparison(name, testType string, post interface{}) {
+	if post == nil { // If nil, test not supporting post state comparison yet
+		fmt.Printf("skipping state comparison json, not supported: %s\n", name)
```

Comment:
probably should use some logger lib

Reply:
change as you see fit (not sure we use any lib in tests)

Reply:
go's built in logger `import "log"` should be enough for this.
If we want to think about a library-wide logging strategy and library to use, it's for sure out of this PR scope.